presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -182,7 +182,60 @@ public boolean isDeterministic()
 
     private boolean validate()
     {
-        controlResult = executeQuery(controlGateway, controlUsername, controlPassword, queryPair.getControl(), controlTimeout, sessionProperties);
+        boolean result = false;
```

Comment:
Thanks for the comment. I actually understood that logic for a query to be deterministic/non-deterministic. It wasn't just quite clear for me what should be the general behavior for a situation when setup query fails, more specifically, if a control setup query fails? Should we rerun it? Should we execute the teardown before rerunning it? Let's discuss this tomorrow and try to go through all the possible flows and then we'll see if we need the deterministic check for setup()/terdown()